### ISSUE TITLE
Add diffing functionality

### ISSUE THREAD

**Issue by Kashomon** [GoogleCloudPlatform/k8s-cluster-bundle#19]
It would be really cool to have diffing functionality for bundles, node configs, components, cluster objects. Not something that is planned for the very near term, but definitely something planned for reasonably soon -- it's essentially for building automation.

**Comment by Zyqsempai** [comment]
@Kashomon So, how do you think, the time is come? I will gladly participate in such feature development.

**Comment by Kashomon** [comment]
Oh yeah, this would be great. Ultimately, I imagine that this would take the form of working on `unstructured.Unstructured` so it could be generic. Honestly, there's probably something that already exists somewhere, since `unstructured.Unstructured` is a standard Kubernetes type.

I'm sort of imagining at the end of the day, the CLI would look like:

`bundlectl diff --diff-files=foo.yaml,bar.yaml`

I think it probably doesn't make sense to diff from STDIN or a mix of STDIN and files.

**Comment by Zyqsempai** [comment]
Yep, definitely, we should rework something instead of wrighting the whole feature from scratch, i found something like this https://github.com/kubernetes/kubernetes/pull/8662/files , but seems like it wasn't merged.

**Comment by Kashomon** [comment]
It's probably good to model it on kubectl as much as possible. Given that the bundle is alpha, I'm happy to restructure the CLI to make it more kubectly-compatible.

**Comment by Kashomon** [comment]
@Zyqsempai -- did you look into this any more?

**Comment by Zyqsempai** [comment]
Yeah, will be glad to help!

**Comment by Kashomon** [comment]
I just noticed https://github.com/sergi/go-diff. That could be quite useful.
